galley/pkg/runtime/strategy.go

```diff
@@ -0,0 +1,124 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package runtime
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	// Maximum wait time before deciding to publish the events.
+	defaultMaxWaitDuration = time.Minute
+
+	// Minimum time distance between two events for deciding on the quiesce point. If the time delay
+	// between two events is larger than this, then we can deduce that we hit a quiesce point.
+	defaultQuiesceDuration = time.Second * 5
+
+	// The frequency for firing the timer events.
+	defaultTimerFrequency = time.Second
+)
+
+// publishingStrategy is a heuristic model for deciding when to publish snapshots. It tries to detect
+// quiesce points for events with a total bounded wait time.
+type publishingStrategy struct {
+	maxWaitDuration time.Duration
+	quiesceDuration time.Duration
+	timerFrequency  time.Duration
+
+	lock sync.Mutex
```

Comment:
A comment about what this lock protects (or grouping fields w/o extra whitespace) would be helpful.